src/backend/distributed/executor/local_executor.c

```diff
@@ -235,6 +237,147 @@ ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterT
 }
 
 
+/*
+ * ExecuteLocalUtilityTaskList executes a list of tasks locally. This function
+ * also logs local execution notice for each task and sets
+ * TransactionAccessedLocalPlacement to true for next set of possible queries
+ * & commands within the current transaction block. See the comment in function.
+ */
+void
+ExecuteLocalUtilityTaskList(List *localTaskList)
+{
+	Task *localTask = NULL;
+
+	foreach_ptr(localTask, localTaskList)
+	{
+		const char *localTaskQueryCommand = TaskQueryString(localTask);
+
+		if (localTask->anchorShardId != INVALID_SHARD_ID)
+		{
+			/*
+			 * If anchor shard id is valid, then we really access a local
+			 * placement. We should register it to force the local execution
+			 * of the following commands withing the current transcation.
+			 * It can be a coordinated transaction or a transaction that is
+			 * initiated explicitly.
+			 */
+			TransactionAccessedLocalPlacement = true;
+		}
+
+		LogLocalCommand(localTask);
+
+		/* execute the local utility task finally */
+		LocallyExecuteUtilityTask(localTaskQueryCommand);
+	}
+}
+
+
+/*
+ * LocallyExecuteUtilityTask executes the given local task query. Given
+ * query would mostly be a utility command as the name of the function
```

Comment:
When will it not be a utility command? If so why will that be executed via this function?

Reply:
We sometimes trigger select udf() `Task`s to complete the execution of some utility commands as the comment within the function implies.